components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -229,127 +230,131 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
+                                        }
 
-                                    if (getThrottleDataHolder().
-                                            isThrottled(combinedResourceLevelThrottleKey)) {
-                                        if (!apiLevelThrottledTriggered) {
-                                            isResourceLevelThrottled = isThrottled = true;
-                                        } else {
-                                            isApiLevelThrottled = isThrottled = true;
+                                        if (getThrottleDataHolder().
+                                                isThrottled(combinedResourceLevelThrottleKey)) {
+                                            if (!apiLevelThrottledTriggered) {
+                                                isResourceLevelThrottled = isThrottled = true;
+                                            } else {
+                                                isApiLevelThrottled = isThrottled = true;
+                                            }
+                                            long timestamp = getThrottleDataHolder().
+                                                    getThrottleNextAccessTimestamp(combinedResourceLevelThrottleKey);
+                                            synCtx.setProperty(APIThrottleConstants.THROTTLED_NEXT_ACCESS_TIMESTAMP,
+                                                    timestamp);
                                         }
-                                        long timestamp = getThrottleDataHolder().
-                                                getThrottleNextAccessTimestamp(combinedResourceLevelThrottleKey);
-                                        synCtx.setProperty(APIThrottleConstants.THROTTLED_NEXT_ACCESS_TIMESTAMP,
-                                                timestamp);
                                     }
-                                }
-                            }else{
-                                if (log.isDebugEnabled()) {
-                                    log.debug("Evaluating Conditional Groups");
-                                }
-                                //Then we will apply resource level throttling
-                                List<ConditionGroupDTO> applicableConditions = getThrottleConditionEvaluator()
-                                        .getApplicableConditions(synCtx, authContext, conditionGroupDTOs);
-                                for (ConditionGroupDTO conditionGroup : applicableConditions) {
-                                    String combinedResourceLevelThrottleKey = resourceLevelThrottleKey + conditionGroup.getConditionGroupId();
-
+                                } else {
                                     if (log.isDebugEnabled()) {
-                                        log.debug("Checking condition : " + combinedResourceLevelThrottleKey);
+                                        log.debug("Evaluating Conditional Groups");
```

Comment:
Shall we include some details regarding the conditional groups evaluated? else the details of this debug log is not much insightful.